src/tls_builtin.c

```diff
@@ -1684,15 +1689,22 @@ void mg_tls_free(struct mg_connection *c) {
 long mg_tls_send(struct mg_connection *c, const void *buf, size_t len) {
   struct tls_data *tls = (struct tls_data *) c->tls;
   long n = MG_IO_WAIT;
-  if (len > MG_IO_SIZE) len = MG_IO_SIZE;
-  if (len > 16384) len = 16384;
-  mg_tls_encrypt(c, (const uint8_t *) buf, len, MG_TLS_APP_DATA);
+  bool was_throttled = c->is_throttled;  // see #3074
```

Comment:
non vc98 friendly, please test  this PR on vc98 with TLS_BUILTIN

Reply:
Nope, that comes from replacing ntohs ntohl with inlines...